libs/providers/go-feature-flag/src/lib/go-feature-flag-provider.ts

```diff
@@ -9,40 +9,132 @@ import {
   TypeMismatchError,
 } from '@openfeature/js-sdk';
 import axios from 'axios';
-import { transformContext } from './context-transformer';
-import { ProxyNotReady } from './errors/proxyNotReady';
-import { ProxyTimeout } from './errors/proxyTimeout';
-import { UnknownError } from './errors/unknownError';
-import { Unauthorized } from './errors/unauthorized';
+import {transformContext} from './context-transformer';
+import {ProxyNotReady} from './errors/proxyNotReady';
+import {ProxyTimeout} from './errors/proxyTimeout';
+import {UnknownError} from './errors/unknownError';
+import {Unauthorized} from './errors/unauthorized';
+import {LRUCache} from 'lru-cache';
 import {
+  DataCollectorRequest,
+  DataCollectorResponse,
+  FeatureEvent,
   GoFeatureFlagProviderOptions,
   GoFeatureFlagProxyRequest,
   GoFeatureFlagProxyResponse,
   GoFeatureFlagUser,
 } from './model';
 
-
 // GoFeatureFlagProvider is the official Open-feature provider for GO Feature Flag.
 export class GoFeatureFlagProvider implements Provider {
   metadata = {
     name: GoFeatureFlagProvider.name,
   };
 
   // endpoint of your go-feature-flag relay proxy instance
-  private endpoint: string;
+  private readonly endpoint: string;
+
   // timeout in millisecond before we consider the request as a failure
-  private timeout: number;
-  // apiKey contains the token to use while calling GO Feature Flag relay proxy
-  private apiKey?: string;
+  private readonly timeout: number;
+
+  // cache contains the local cache used in the provider to avoid calling the relay-proxy for every evaluation
+  private readonly cache?: LRUCache<string, ResolutionDetails<any>>;
+
+  // bgSchedulerId contains the id of the setInterval that is running.
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  private bgScheduler?: NodeJS.Timer;
+
+  // dataCollectorBuffer contains all the FeatureEvents that we need to send to the relay-proxy for data collection.
+  private dataCollectorBuffer?: FeatureEvent<any>[];
+
+  // dataCollectorMetadata are the metadata used when calling the data collector endpoint
+  private readonly dataCollectorMetadata: Record<string, string> = {
+    provider: 'open-feature-js-sdk',
+  };
+
+  // cacheTTL is the time we keep the evaluation in the cache before we consider it as obsolete.
+  // If you want to keep the value forever you can set the FlagCacheTTL field to -1
+  private readonly cacheTTL?: number;
```

Comment:
Just some thoughts:

LRU will automatically delete oldest items when it reaches it's specified size.So you can avoid a flush and TTL entirely unless you are worried about stale flag values.

If you _are_ worried about stale flag values, a better solution may be some way of invalidating the cache when a change is detected upstream, but I'm not sure of GoFF supports that.

Reply:
GO feature flag is not supporting it for now, so the TTL is needed in case of stale flag values.

I am actually working on it in this PR https://github.com/thomaspoignant/go-feature-flag/pull/802 but it is not ready yet.